### PR TITLE
pkg tests fix for macos

### PR DESCRIPTION
**Pull request by Akm0d**
### What does this PR do?
Stop the mac tests from holding up
The issue may have been created here: https://github.com/saltstack/salt/pull/54735/files

### CODE REVIEW

**Comment by Akm0d** [comment]
The infinite recursion stems from the `requires_salt_modules` wrapper.
it calls `salt/utils/mac_utils:available_services` infinitely

**Comment by Akm0d** [comment]
re-run full macosxmojave

**Comment by Akm0d** [comment]
These log lines are on repeat for 6 hours until the test times out 
```
16:14:01,074 [salt.loader                                               :747 ][DEBUG   ] Grains refresh requested. Refreshing grains.
16:14:01,075 [salt.config                                               :2185][DEBUG   ] Reading configuration from /private/tmp/salt-tests-tmpdir/config/syndic-master/master
16:14:01,949 [salt.utils.lazy                                           :104 ][DEBUG   ] LazyLoaded local_cache.clean_old_jobs
16:14:02,027 [salt.utils.lazy                                           :104 ][DEBUG   ] LazyLoaded localfs.list_tokens
16:14:02,029 [salt.utils.verify                                         :456 ][DEBUG   ] This salt-master instance has accepted 1 minion keys.
16:14:10,219 [salt.master                                               :445 ][DEBUG   ] Performing fileserver updates for items with an update interval of 60
16:14:10,221 [salt.master                                               :457 ][DEBUG   ] Updating roots fileserver cache
16:14:10,240 [salt.master                                               :469 ][DEBUG   ] Completed fileserver updates for items with an update interval of 60, waiting 60 seconds
16:14:51,993 [salt.config                                               :2185][DEBUG   ] Reading configuration from /private/tmp/salt-tests-tmpdir/config/master
16:14:51,997 [salt.config                                               :2329][DEBUG   ] Including configuration from '/private/tmp/salt-tests-tmpdir/config/minion.d/_schedule.conf'
16:14:51,998 [salt.config                                               :2185][DEBUG   ] Reading configuration from /private/tmp/salt-tests-tmpdir/config/minion.d/_schedule.conf```
Because of the changes that caused this issue I'm starting to suspect using `@requires_system_grains` on a `setUpClass` method as well as on individual test functions in the same class is the culprit

**Comment by waynew** [comment]
Is the grains refresh line the first one in there?

**Comment by Akm0d** [comment]
@waynew  yes

**Comment by Akm0d** [comment]
Skipping `test_user_delete_kw_output` in `test_call` solves the mac issue.  I see no reason why the test shouldn't work, will revisit it later and I'm going to merge the test skip for now so we aren't blocked by it

**Comment by Akm0d** [comment]
reverting: https://github.com/saltstack/salt/pull/54735/commits
Cherry-picking https://github.com/saltstack/salt/pull/54851/commits

**Comment by Akm0d** [comment]
> reverting: https://github.com/saltstack/salt/pull/54735/commits
> Cherry-picking https://github.com/saltstack/salt/pull/54851/commits

Tests still failed

**Review by twangboy — APPROVED** [review]

File: .ci/kitchen-macosxmojave-py3
Comment (Akm0d) [reply .ci/kitchen-macosxmojave-py3]: That's part of the commit labeled `undo this commit before we merge`.  Once the tests all pass I'll revert that commit.

**Comment by Akm0d** [comment]
re-run full windows2019

**Comment by Akm0d** [comment]
Getting this on Macos Py3, but it's not failing any tests, let's look at that another day
```21:03:39           test_package (unit.test_loader.LazyLoaderModulePackageTest) ... 21:03:38,931 [salt.loader                                                 :1751][ERROR   ] Failed to load function mac_utils.available_services because its module (mac_utils) is not in the whitelist: ['boto', 'boto3', 'args', 'systemd', 'path', 'platform', 'reg']```

**Comment by Akm0d** [comment]
re-run full windows

**Comment by Akm0d** [comment]
re-run full macosxmojave

**Comment by Akm0d** [comment]
re-run full debian10

**Review by dwoz — APPROVED** [review]

**Review by waynew — APPROVED** [review]

**Pull request merged**
